scripts/tests/test_safe_template_linter.py

```diff
@@ -721,12 +733,197 @@ def test_check_underscore_file_disables_one_violation(self):
     )
     def test_check_underscore_no_escape_allowed(self, data):
         """
-        Test _check_underscore_file_is_safe with expressions that are allowed
+        Test check_underscore_file_is_safe with expressions that are allowed
         without escaping because the internal calls properly escape.
         """
         linter = UnderscoreTemplateLinter()
         results = FileResults('')
 
-        linter._check_underscore_file_is_safe(data['template'], results)
+        linter.check_underscore_file_is_safe(data['template'], results)
 
         self.assertEqual(len(results.violations), 0)
+
+
+@ddt
+class TestJavaScriptLinter(TestLinter):
+    """
+    Test JavaScriptLinter
+    """
+
+    @data(
+        {'template': 'var m = "Text " + message + " text."', 'rule': None},
+        {'template': 'var m = "<p>" + message + "</p>"', 'rule': Rules.javascript_concat_html},
+        {'template': 'var m = " <p> " + message + " </p> "', 'rule': Rules.javascript_concat_html},
+    )
+    def test_concat_with_html(self, data):
+        """
+        Test _check_javascript_file_is_safe with concatenating strings and HTML
+        """
+        linter = JavaScriptLinter()
+        results = FileResults('')
+
+        linter._check_javascript_file_is_safe(data['template'], results)
+        self._validate_data_rule(data, results)
+
+    @data(
+        {'template': 'test.append( test.render().el )', 'rule': None},
+        {'template': 'test.append( test.render().$el )', 'rule': None},
+        {'template': 'test.append( testEl )', 'rule': None},
+        {'template': 'test.append( $test )', 'rule': None},
+        {'template': 'test.append( "text" )', 'rule': None},
+        {'template': 'test.append( $( "<div>" ) )', 'rule': None},
+        {'template': 'test.append( $( "<div/>" ) )', 'rule': None},
+        {'template': 'test.append(HtmlUtils.forJQuery(someHtml))', 'rule': None},
+        {'template': 'HtmlUtils.append($el, someHtml)', 'rule': None},
+        {'template': 'test.append("fail on concat" + test.render().el)', 'rule': Rules.javascript_jquery_append},
+        {'template': 'test.append("fail on concat" + testEl)', 'rule': Rules.javascript_jquery_append},
+        {'template': 'test.append( message )', 'rule': Rules.javascript_jquery_append},
+    )
+    def test_jquery_append(self, data):
+        """
+        Test _check_javascript_file_is_safe with JQuery append()
+        """
+        linter = JavaScriptLinter()
+        results = FileResults('')
+
+        linter._check_javascript_file_is_safe(data['template'], results)
+
+        self._validate_data_rule(data, results)
+
+    @data(
+        {'template': 'test.prepend( test.render().el )', 'rule': None},
```

Comment:
@andy-armstrong: I added linting of CoffeeScript too.  Is there anything like `@el` that should be exempt?  Maybe you could add a comment with some tests that you think should pass in CoffeeScript.  This may apply to some other tests as well.  I don't know CoffeeScript.  Thanks!